### ISSUE TITLE
Reading flex.symfony.com on composer install when composer.lock and symfony.lock files are present

### ISSUE THREAD

**Issue by bbaga** [symfony/flex#527]
We are facing a flex related problem, it looks like flex is trying to fetch the `versions.json` during `composer install`, even though the lock files are present. Our workflow and setup is nothing special, composer packages mirrored with artifactory, composer.lock committed to the repo and we run `composer install` before creating the deploy artifacts.

We had no problem with this workflow until `1.2.6` was released, then our build pipelines started to fail, because `flex.symfony.com` is unreachable from our build machines (they can access internal resources only).

I was wondering, what is the purpose of that http call. My understanding is that, at that point all necessary package information should be available from the lock files.

`symonfy/flex`: `>1.2.5`

**Comment by bbaga** [comment]
Additionally, we tried to run `composer install --no-plugins`, but then our `post-install-cmd` scripts are failing, supposedly due to composer's scripts plugin getting disabled as well.

This issue forced us to lock flex's version on `1.2.5` and to start investigating mirroring solutions.

**Comment by bitgandtter** [comment]
we are facing the same issue

**Comment by bbaga** [comment]
@bitgandtter this is the only thing which looked like a reasonable project is this: https://github.com/moay/server-for-symfony-flex

We haven't tried it yet, still hoping this is going to be resolved within `flex`.
At the moment we are locked on `1.2.5`.

**Comment by bitgandtter** [comment]
wow a week now without response, I'm not able to use the latest version of flex because of this issue

**Comment by nicolas-grekas** [comment]
@bitgandtter would you mind having a look how to prevent doing this call? I agree it should be preventable and prevented.

**Comment by bitgandtter** [comment]
hello again, so weird, after I give it a try today and after an also weird error message 

`"https://flex.symfony.com/versions.json" does not contain valid JSON
Parse error on line 1:

^
Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
https://flex.symfony.com/versions.json could not be fully loaded, package information was loaded from the local cache and may be out of date
`

it works again with the latest flex version. I also review the latest commits and I don't see any related to it.

**Comment by o-alquimista** [comment]
> hello again, so weird, after I give it a try today and after an also weird error message
> 
> `"https://flex.symfony.com/versions.json" does not contain valid JSON
> Parse error on line 1:
> 
> ^
> Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
> https://flex.symfony.com/versions.json could not be fully loaded, package information was loaded from the local cache and may be out of date
> `
> 
> it works again with the latest flex version. I also review the latest commits and I don't see any related to it.

I'm running the Symfony Demo app in Docker containers and this very same error about versions.json is ruining it.

The funny thing is that the same Docker project builds and executes flawlessly on another computer running the same operating system (Fedora 30). I mean it. The very same OS with the same settings and changes, freshly installed, ending with two different results. I don't understand this anymore!

**Comment by bbaga** [comment]
As an update, I started looking into the code and working on a possible fix.
The issue seems to be `flex` is [fetching the versions much earlier than it is needed](https://github.com/symfony/flex/blob/master/src/Flex.php#L121).

Also noticed that if I'm running the unit tests for `flex` in isolation, one of the tests fails because it can't fetch the versions.
I was able to make all the tests pass with deferring the call until the data is required, will post the PR soon.
